gnd/src/main/java/com/google/android/gnd/ui/offlinearea/viewer/OfflineAreaViewerFragment.java

```diff
@@ -0,0 +1,114 @@
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import butterknife.OnClick;
+import com.google.android.gnd.MainActivity;
+import com.google.android.gnd.R;
+import com.google.android.gnd.databinding.OfflineAreaViewerFragBinding;
+import com.google.android.gnd.model.basemap.OfflineArea;
+import com.google.android.gnd.model.feature.Point;
+import com.google.android.gnd.ui.common.AbstractFragment;
+import com.google.android.gnd.ui.map.MapAdapter;
+import com.google.android.gnd.ui.map.MapProvider;
+import io.reactivex.Single;
+import javax.inject.Inject;
+
+/**
+ * The OfflineAreaViewerFragment provides a UI for managing a single offline area on the user's
+ * device.
+ * */
+public class OfflineAreaViewerFragment extends AbstractFragment {
+
+  private static final String MAP_FRAGMENT = MapProvider.class.getName() + "#fragment";
+
+  @Inject MapProvider mapProvider;
+
+  private OfflineAreaViewerViewModel viewModel;
+  @Nullable private MapAdapter map;
+
+  @Inject
+  public OfflineAreaViewerFragment() {}
+
+  @Override
+  public void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    OfflineAreaViewerFragmentArgs args = OfflineAreaViewerFragmentArgs.fromBundle(getArguments());
+    viewModel = getViewModel(OfflineAreaViewerViewModel.class);
+    viewModel.loadOfflineArea(args);
+    Single<MapAdapter> mapAdapter = mapProvider.getMapAdapter();
+    mapAdapter.as(autoDisposable(this)).subscribe(this::onMapReady);
+    viewModel.getOfflineArea().observe(this, this::panMap);
+  }
+
+  @Override
+  public View onCreateView(
+      @NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+    super.onCreateView(inflater, container, savedInstanceState);
+    OfflineAreaViewerFragBinding binding =
+        OfflineAreaViewerFragBinding.inflate(inflater, container, false);
+    binding.setViewModel(viewModel);
+    binding.setLifecycleOwner(this);
+    ((MainActivity) getActivity()).setActionBar(binding.offlineAreaViewerToolbar, true);
+    return binding.getRoot();
+  }
+
+  @Override
+  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+    if (savedInstanceState == null) {
+      replaceFragment(R.id.map, mapProvider.getFragment());
+    } else {
+      mapProvider.restore(restoreChildFragment(savedInstanceState, MAP_FRAGMENT));
+    }
+  }
+
+  private void onMapReady(MapAdapter map) {
+    this.map = map;
+  }
+
+  private void panMap(OfflineArea offlineArea) {
+    if (map == null) {
+      return;
+    }
+
+    double lat = offlineArea.getBounds().northeast.latitude;
+    double lon = offlineArea.getBounds().southwest.longitude;
+    Point point = Point.newBuilder().setLatitude(lat).setLongitude(lon).build();
+    map.moveCamera(point);
+  }
+
+  /**
+   * Removes the area associated with this fragment from the user's device.
+   * */
+  @OnClick(R.id.remove_button)
```

Comment:
Since butterknife is already deprecated, we should prevent adding more code that needs to be cleaned up later. Can you please fix this?